internal/auth/authenticators_test.go

```diff
@@ -0,0 +1,143 @@
+package auth_test
+
+import (
+	"encoding/hex"
+	"net/http"
+	"testing"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	compat "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	ec "github.com/bitcoin-sv/go-sdk/primitives/ec"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/auth"
+	"github.com/go-resty/resty/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestXpubAuthenticator_Authenticate(t *testing.T) {
+	key := extendedKeyTestHelper(t)
+	authenticator, err := auth.NewXpubOnlyAuthenticator(key)
+	require.NotNil(t, authenticator)
+	require.NoError(t, err)
+
+	req := resty.New().R()
+	err = authenticator.Authenticate(req)
+	require.NoError(t, err)
+	xPubHeadersTestHelper(t, req.Header, key)
```

Comment:
Please add the section given/when/then to all tests (we have even a section for that in [code standards](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md#guidelines-for-writing-tests) )

At least something like that:
```suggestion
        // given:
	key := extendedKeyTestHelper(t)
	
	authenticator, err := auth.NewXpubOnlyAuthenticator(key)
	require.NotNil(t, authenticator)
	require.NoError(t, err)

	req := resty.New().R()
	
	 // when:
	err = authenticator.Authenticate(req)
	
	// then:
	require.NoError(t, err)
	xPubHeadersTestHelper(t, req.Header, key)
```